### PR TITLE
Sig 2359 cypress regression test

### PR DESCRIPTION
**Pull request by arjandata**
## Cypress

This pull request adds Cypress to are code base for regression testing

### CODE REVIEW

File: environment.conf.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
+  "API_ROOT": "http://localhost:8000/",
   "ROOT": "http://localhost:3001/",
```
Comment (jpoppe) [inline environment.conf.json]: I don't think those lines should be pushed ;)

File: cypress/fixtures/wespen.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    ],
+    "subrubriek": [
+        [
+            "https://acc.api.data.amsterdam.nl/signals/v1/public/terms/categories/overlast-van-dieren/sub_categories/wespen"
+        ],
+        [
+            0.7382120726922458
+        ]
+    ]
+}
```
Comment (jpoppe) [inline cypress/fixtures/wespen.json]: Those seem to be Windows carriage returns and should be removed.

File: cypress.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    "baseUrl": "http://localhost:3001",
+    "waitForAnimations": true,
+    "animationsDistanceTreshold": 20,
+    "defaultCommandTimeOut": 5000,
+    "execTimeout": 60000,
+    "pageLoadTimeout": 60000,
+    "requestTimeout": 15000,
+    "reponsTimeout": 15000,
+    "failOnStatusCode": false
+}
```
Comment (jpoppe) [inline cypress.json]: Those seem to be Windows carriage returns and should be removed (they are in multiple files)

File: environment.conf.json (hunk 1 above)
Comment (jpoppe) [reply environment.conf.json]: You also need to revert the next line ;)

File: cypress/integration/filters.spec.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    // Click on 4th column stadsdeel to sort ASC
+    cy.get('th').eq(3).should('contain', 'Stadsdeel').click();
+    cy.wait('@getSortedASC');
+    cy.get('th.sort.sort-up').should('be.visible');
+
+    // After sorting, first element should contain stadsdeel Centrum
+    cy.get(MANAGE_SIGNALS.firstSignalStadsdeelName).should('contain', 'Centrum');
+
+    // Click on 4th column stadsdeel to sort DESC
+    cy.get('th').eq(3).should('contain', 'Stadsdeel').click();
```
[the remark this answers is not in the document]
Comment (arjandata) [reply cypress/integration/filters.spec.js:59]: This is how i check the results of the filtering.  In both cases the first line should have the same 'stadsdeel' name.

File: cypress/support/commandsGeneral.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+});
+// go to the page, use delete win.fetch to convert fetch requests in xhr
+Cypress.Commands.add('visitFetch', url =>{
+  cy.visit(url,{
+    onBeforeLoad(win) {
+      // eslint-disable-next-line no-param-reassign
+      delete win.fetch;
+    },
+  });
+});
```
[the remark this answers is not in the document]
Comment (arjandata) [reply cypress/support/commandsGeneral.js:51]: Cypress cannot intercept the fetch protocol, this is a workaround to intercept it and fall back to the XmlHttpRequest protocol. I added this comment to the code.

File: cypress/integration/serviceBeloften.spec.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    // Wait for saving the data
+    cy.wait('@patchCategory');
+
+    // Check if Categorieën page opens again
+    cy.url().should('include', '/instellingen/categorieen/page/1');
+    cy.checkHeader('Categorieën');
+
+    // Check day change
+    cy.reload(true);
+    cy.get('[data-testid=dataViewBody] > [data-testid=dataViewBodyRow]',{ timeout: 10000 }).first().contains('4 dagen');
```
[the remark this answers is not in the document]
Comment (arjandata) [reply cypress/integration/serviceBeloften.spec.js:66]: It takes some time for the GUI to update. The default timeout value is not enough.

File: cypress/integration/filters.spec.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    cy.wait('@getSortedDESC');
+    cy.get('th.sort.sort-down').should('be.visible');
+    cy.wait(1000); // eslint-disable-line cypress/no-unnecessary-waiting
```
[the remark this answers is not in the document]
Comment (arjandata) [reply cypress/integration/filters.spec.js:62]: I know this is not a proper way of working. But I see no other solution. I wait for the results after sorting with @getSortedDESC. There is no other request to wait for. Without the wait of 1000 ms, the test sometimes fails.

File: cypress/integration/filters.spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (janjaap) [reply cypress/integration/filters.spec.js:59]: I understand, but should that be part of this test case? If the sorting fails, then the case, which is unrelated to sorting, fails, which gives a false notion of failure.

**Pull request merged**